woocommerce/woocommerce
# render svg suggestion icons from json API content

### Changes proposed in this Pull Request:

This PR renders SVG icons from marketplace suggestions API content, instead of using urls pointing to (external) images.

Using SVGs embedded in the API response means that customer sites won't need to make additional image requests to WCCOM or elsewhere for these icon images.

Note that we're directly rendering the returned SVG into the dom with `Element.innerHTML`. This implies a reasonable level of trust between woo core <=> wccom.

### How to test the changes in this Pull Request:

1. Clone this branch in a woo dev environment.
2. Replace `icon` urls in suggestion content with SVG content (see below for example svg). You'll need to make sure this json data is used. Ensure suggestion content is loaded & cached (`wp option get woocommerce_marketplace_suggestions`). One way to use custom data is to hijack the http request to the API (`pre_http_request`), clear the cache (option) using CLI, then use `wp shell` to add a job, and then trigger cron/action scheduler to run that job. 😅 
4. View suggestions in various contexts – should see your SVG image.

```json
"icon": "<svg width=\"40\" height=\"40\" viewBox=\"0 0 32 32\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\"><mask id=\"mask0\" mask-type=\"alpha\" maskUnits=\"userSpaceOnUse\" x=\"0\" y=\"0\" width=\"32\" height=\"32\"><rect width=\"32\" height=\"32\" rx=\"16\" fill=\"#9B5C8F\"/></mask><g mask=\"url(#mask0)\"><rect width=\"32\" height=\"32\" rx=\"16\" fill=\"#9B5C8F\"/><path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M17.1094 4.10919C16.9516 4.26697 16.9481 4.51932 17.1016 4.67282C17.1016 4.67282 17.9353 5.50656 17.9275 6.07019C17.9197 6.63382 17.0626 7.49097 17.0626 7.49097C16.9048 7.64875 16.9013 7.9011 17.0548 8.0546C17.2083 8.20809 17.4606 8.20458 17.6184 8.0468C17.6184 8.0468 18.4756 7.18966 19.0392 7.18186C19.6028 7.17406 20.4366 8.0078 20.4366 8.0078C20.5901 8.1613 20.8424 8.15779 21.0002 8C21.158 7.84222 21.1615 7.58987 21.008 7.43637C21.008 7.43637 20.1713 6.81721 20.182 6.039C20.1928 5.26079 21.047 4.61822 21.047 4.61822C21.2048 4.46044 21.2083 4.20809 21.0548 4.05459C20.9013 3.9011 20.6489 3.90461 20.4912 4.06239C20.4912 4.06239 19.634 4.91954 19.0704 4.92734C18.5068 4.93514 17.673 4.10139 17.673 4.10139C17.5195 3.9479 17.2672 3.9514 17.1094 4.10919Z\" fill=\"#F1D79D\"/><path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M11.0018 7.27379C10.8655 7.27379 10.7549 7.3843 10.7549 7.52063C10.7549 7.52063 10.7549 8.26114 10.5081 8.50797C10.2613 8.75481 9.52076 8.75481 9.52076 8.75481C9.38445 8.75481 9.27393 8.86532 9.27393 9.00165C9.27393 9.13798 9.38445 9.24848 9.52076 9.24848C9.52076 9.24848 10.2613 9.24848 10.5081 9.49532C10.7549 9.74216 10.7549 10.4827 10.7549 10.4827C10.7549 10.619 10.8655 10.7295 11.0018 10.7295C11.1381 10.7295 11.2486 10.619 11.2486 10.4827C11.2486 10.4827 11.1546 9.83613 11.4955 9.49532C11.8363 9.15451 12.4828 9.24848 12.4828 9.24848C12.6191 9.24848 12.7296 9.13798 12.7296 9.00165C12.7296 8.86532 12.6191 8.75481 12.4828 8.75481C12.4828 8.75481 11.7423 8.75481 11.4955 8.50797C11.2486 8.26114 11.2486 7.52063 11.2486 7.52063C11.2486 7.3843 11.1381 7.27379 11.0018 7.27379Z\" fill=\"white\"/><path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M22.2218 12.2218C22.1254 12.3182 22.1254 12.4745 22.2218 12.5709C22.2218 12.5709 22.7454 13.0945 22.7454 13.4436C22.7454 13.7926 22.2218 14.3163 22.2218 14.3163C22.1254 14.4126 22.1254 14.5689 22.2218 14.6653C22.3182 14.7617 22.4745 14.7617 22.5709 14.6653C22.5709 14.6653 23.0945 14.1417 23.4436 14.1417C23.7926 14.1417 24.3163 14.6653 24.3163 14.6653C24.4127 14.7617 24.5689 14.7617 24.6653 14.6653C24.7617 14.5689 24.7617 14.4127 24.6653 14.3163C24.6653 14.3163 24.1417 13.9255 24.1417 13.4436C24.1417 12.9616 24.6653 12.5709 24.6653 12.5709C24.7617 12.4745 24.7617 12.3182 24.6653 12.2218C24.5689 12.1254 24.4126 12.1254 24.3163 12.2218C24.3163 12.2218 23.7926 12.7454 23.4436 12.7454C23.0945 12.7454 22.5709 12.2218 22.5709 12.2218C22.4745 12.1254 22.3182 12.1254 22.2218 12.2218Z\" fill=\"white\"/><path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M20.8391 17C20.6151 17 20.4335 17.1816 20.4335 17.4056C20.4335 17.4056 20.4335 18.6223 20.0279 19.0279C19.6223 19.4335 18.4056 19.4335 18.4056 19.4335C18.1816 19.4335 18 19.6151 18 19.8391C18 20.0631 18.1816 20.2447 18.4056 20.2447C18.4056 20.2447 19.6223 20.2447 20.0279 20.6503C20.4335 21.0559 20.4335 22.2726 20.4335 22.2726C20.4335 22.4966 20.6151 22.6782 20.8391 22.6782C21.0631 22.6782 21.2447 22.4966 21.2447 22.2726C21.2447 22.2726 21.0903 21.2103 21.6503 20.6503C22.2103 20.0903 23.2726 20.2447 23.2726 20.2447C23.4966 20.2447 23.6782 20.0631 23.6782 19.8391C23.6782 19.6151 23.4966 19.4335 23.2726 19.4335C23.2726 19.4335 22.0559 19.4335 21.6503 19.0279C21.2447 18.6223 21.2447 17.4056 21.2447 17.4056C21.2447 17.1816 21.0631 17 20.8391 17Z\" fill=\"#F1D79D\"/><path d=\"M15.8665 12.1135C16.3825 11.4985 17.2993 11.4183 17.9143 11.9343C18.5292 12.4503 18.6094 13.3672 18.0934 13.9821L-1.83872 37.7363C-2.35473 38.3513 -3.27156 38.4315 -3.88652 37.9155C-4.50148 37.3995 -4.58169 36.4826 -4.06568 35.8677L15.8665 12.1135Z\" fill=\"#E9DDE7\"/><path d=\"M18.0934 13.9821C18.6094 13.3672 18.5292 12.4503 17.9143 11.9343C17.2993 11.4183 16.3825 11.4985 15.8665 12.1135L12.0842 16.621L14.3806 18.4069L18.0934 13.9821Z\" fill=\"white\"/><path d=\"M-1.83892 37.7363L18.0932 13.9821C18.6092 13.3672 18.529 12.4503 17.9141 11.9343L-3.88672 37.9155C-3.27176 38.4315 -2.35493 38.3513 -1.83892 37.7363Z\" fill=\"#9B5C8F\" fill-opacity=\"0.2\"/></g></svg>",
```

### Screenshots

<img width="843" alt="screen shot 2019-02-28 at 12 04 40 pm" src="https://user-images.githubusercontent.com/4167300/53529308-10196400-3b51-11e9-879b-762fa77e1739.png">

<img width="697" alt="screen shot 2019-02-28 at 12 04 49 pm" src="https://user-images.githubusercontent.com/4167300/53529309-13145480-3b51-11e9-9213-c61375ae35ec.png">

<img width="1078" alt="screen shot 2019-02-28 at 12 04 21 pm" src="https://user-images.githubusercontent.com/4167300/53529316-17d90880-3b51-11e9-8bdb-de05a783537a.png">
